### PR TITLE
[1.16] Update Golang to v1.13.14

### PR DESCRIPTION
**Pull request by justaugustus**
**What type of PR is this?**

/kind cleanup
/area dependency release-eng
/sig testing release

**What this PR does / why we need it**:

Update Golang to v1.13.14.

Tracking issue: https://github.com/kubernetes/release/issues/1410

**Which issue(s) this PR fixes**:

**Special notes for your reviewer**:

~Includes the bazel updates from https://github.com/kubernetes/kubernetes/pull/93219.
If this goes green without issue, I'll close the previous PR and update the release note here instead.~

Combined cherry pick of https://github.com/kubernetes/kubernetes/pull/89430, https://github.com/kubernetes/kubernetes/pull/90347, https://github.com/kubernetes/kubernetes/pull/91312, bddd52046d8122af98eab4a172cebb45066a78d1, and https://github.com/kubernetes/kubernetes/pull/92362.

**Does this PR introduce a user-facing change?**:

```release-note
- Update Golang to v1.13.14
- Update bazel to 2.2.0
- Update repo-infra to 0.0.8 (to support go1.14.6 and go1.13.14)
  - Includes:
    - bazelbuild/bazel-toolchains@3.4.0
    - bazelbuild/rules_go@v0.22.8
```

### CODE REVIEW

**Comment by justaugustus** [comment]
`pull-kubernetes-e2e-gce-device-plugin-gpu` is a known flake.
/skip

**Comment by justaugustus** [comment]
Ready for review.
/assign @liggitt @dims @BenTheElder 
cc: @kubernetes/release-engineering

**Comment by liggitt** [comment]
/cc @mikedanese @fejta 
can you ack the bazel changes?

File: build/root/.bazelrc [hunk 1]
```diff
@@ -40,19 +40,18 @@ build:cross:darwin_amd64 --platforms=@io_bazel_rules_go//go/toolchain:darwin_amd
...
 
-# We enable cgo cross-compilation for Linux, but need to use our custom crosstool.
-build:repo_infra_crosstool --crosstool_top=@io_k8s_repo_infra//tools:toolchain --compiler=gcc
-build:cross:linux_386 --config=repo_infra_crosstool --platforms=@io_bazel_rules_go//go/toolchain:linux_386
-build:cross:linux_amd64 --config=repo_infra_crosstool --platforms=@io_bazel_rules_go//go/toolchain:linux_amd64 --cpu=amd64
-build:cross:linux_arm --config=repo_infra_crosstool --platforms=@io_bazel_rules_go//go/toolchain:linux_arm --cpu=arm
-build:cross:linux_arm64 --config=repo_infra_crosstool --platforms=@io_bazel_rules_go//go/toolchain:linux_arm64 --cpu=arm64
-build:cross:linux_ppc64le --config=repo_infra_crosstool --platforms=@io_bazel_rules_go//go/toolchain:linux_ppc64le --cpu=ppc64le
-build:cross:linux_s390x --config=repo_infra_crosstool --platforms=@io_bazel_rules_go//go/toolchain:linux_s390x --cpu=s390x
+# These cross-compilation targets are most likely non-functional.
```
Comment (liggitt) [inline build/root/.bazelrc:43]: > These cross-compilation targets are most likely non-functional.

that's ominous...

**Comment by liggitt** [comment]
verified this matches master bazel bits, and CI passes

/lgtm
/approve

**Review by hasheddan — APPROVED** [review]
/lgtm

**Comment by fejta** [comment]
LGTM

**Pull request merged**